src/ProjectModule/Components/ProjectForm/ProjectForm.tsx

```diff
@@ -0,0 +1,114 @@
+import React, { useEffect, useState } from 'react'
+import styles from './ProjectForm.module.css'
+import { useNavigate, useParams } from 'react-router-dom'
+import { useForm } from 'react-hook-form';
+import axios from 'axios';
+import { baseUrl } from '../../../Constants/Components/Urls';
+import { toast } from 'react-toastify';
+import Loader from "../../../SharedMoudule/Components/Loading/Loading";
+
+
+export default function ProjectForm() {
+    const navigate=useNavigate();
+    const {register,handleSubmit,formState:{errors}}=useForm();
+    const token=localStorage.getItem('userToken');
+    const {action}=useParams();
+    const [currentProject,setCurrentProject]=useState(null);
+    const curruntProjectId=localStorage.getItem('curruntProjectId');
+    const [isLoading, setIsLoading] = useState(false);
+
+    const onSubmitAdd= async(data:any)=>{
+        setIsLoading(true);
+        // console.log('Add');
+        // console.log(data);
+        try {
+            let response=await axios.post(`${baseUrl}/Project`,data,{headers:{Authorization:token}});
+            console.log('add project response ',response);
+            toast.success('Project Added Successfuly');
+            navigate('/dashboard/project-list');
+        } catch (error) {
+            setIsLoading(false);
+            console.log(error);
+            toast.error(error?.response?.data?.message||'Somthing went wrong!');
+        }
+        
+    }
+
+    const onSubmitUpdate= async(data:any)=>{
+        setIsLoading(true);
+        // console.log('Update data');
+        // console.log(data);
+        try {
+            let response=await axios.put(`${baseUrl}/Project/${curruntProjectId}`,{'title':data.title?data.title:currentProject.title,'description':data.description?data.description:currentProject.description},{headers:{Authorization:token}});
+            console.log('update project response ',response);
+            toast.success('Project Updated Successfuly');
+            navigate('/dashboard/project-list');
+        } catch (error) {
+            setIsLoading(false);
+            console.log(error);
+            toast.error(error?.response?.data?.message||'Somthing went wrong!');
+        }
+        
+    }
+
+    const getCurrentProject=async()=>{
+        try {
+            let response=await axios.get(`${baseUrl}/Project/${curruntProjectId}`,{headers:{Authorization:token}});
+            console.log('current project ',response.data);
+            setCurrentProject(response.data);  
```

Comment:
You can get the setValue function from the `useForm` hook  and change the code to the following:
```
const currentProject = response?.data;
setValue("title", currentProject.title);
setValue("description", currentProject.description);
```

The current problem is that we are setting the value using the `defaultValue` but react-hook-form doesn't know anything about it unless we change the value inside the input.
Using the `setValue` function will make the `useForm`  aware of the changes. We can also now remove the currentProject state.